softlayer/baremetal/baremetal_finder_test.go

```diff
@@ -0,0 +1,40 @@
+package baremetal_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	boshlog "github.com/cloudfoundry/bosh-agent/logger"
+	common "github.com/maximilien/bosh-softlayer-cpi/common"
+	bm "github.com/maximilien/bosh-softlayer-cpi/softlayer/baremetal"
+	fakeslclient "github.com/maximilien/softlayer-go/client/fakes"
+)
+
+var _ = Describe("BaremetalFinder", func() {
+	var (
+		softLayerClient *fakeslclient.FakeSoftLayerClient
+		logger          boshlog.Logger
+		finder          bm.BaremetalFinder
+	)
+
+	BeforeEach(func() {
+		softLayerClient = fakeslclient.NewFakeSoftLayerClient("fake-username", "fake-api-key")
+		logger = boshlog.NewLogger(boshlog.LevelNone)
+
+		finder = bm.NewBaremetalFinder(
+			softLayerClient,
+			logger,
+		)
+		common.SetTestFixtureForFakeSoftLayerClient(softLayerClient, "SoftLayer_Hardware_Service_getObject.json")
+	})
+
+	Describe("Find", func() {
```

Comment:
Same here we should have tests for negative paths, e.g., not able to find BM servers and making sure error is returned and propagated.